### PR TITLE
Fix messing research queue exploit in jail

### PR DESCRIPTION
**Pull request by Ragnarok77-factorio**
Fix messing research queue exploit in jail

### Brief description of the changes:
If a player has its research window opened, and if he's jailed, he can still mess research even without permission as the tab is opened. To prevent them, kill them so they loose the window.

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

**Comment by TheBigZet** [comment]
When in doubt, kill the offender! 😄

**Comment by TheBigZet** [comment]
Actually there's non-lethal way to do this
`player.opened = none`
Killing the player would mean the corpse is left forever in jail

File: utils/datastore/jail_data.lua [hunk 1]
```diff
@@ -378,6 +378,7 @@ local jail = function(player, griefer, msg)
 
     game.players[griefer].clear_console()
     Utils.print_to(griefer, message)
+	game.players[griefer].character.die()
```
Comment (TheBigZet) [inline utils/datastore/jail_data.lua:381]: Actually there's non-lethal way to do this @Ragnarok77-factorio
player.opened = none
Killing the player would mean the corpse is left forever in jail

**Review by TheBigZet — CHANGES_REQUESTED** [review]

File: utils/datastore/jail_data.lua (hunk 1 above)
Comment (Ragnarok77-factorio) [reply utils/datastore/jail_data.lua]: Updated it to your request, tested it with 2 accounts (one jailing, the other one having research opened) and... it worked, research closed once jailed. I didn't know this method existed.

**Comment by Ragnarok77-factorio** [comment]
Also added changelogs, seems like it has not been started for a long time

File: utils/datastore/jail_data.lua [hunk 2]
```diff
@@ -378,6 +378,7 @@ local jail = function(player, griefer, msg)
 
     game.players[griefer].clear_console()
     Utils.print_to(griefer, message)
+    game.players[griefer].character.opened = none
```
Comment (TheBigZet) [inline utils/datastore/jail_data.lua:381]: Test it, I believe `opened` belongs to `player`, not `character`

Comment (Ragnarok77-factorio) [reply utils/datastore/jail_data.lua]: Seems to work for both :  https://lua-api.factorio.com/latest/LuaControl.html#LuaControl.opened
image

I tested with or without character, both OK
Anyways, shorter code I guess without character , so updated it @TheBigZet

**Review by TheBigZet — APPROVED** [review]
Technically not the correct syntax, but it works

File: utils/datastore/jail_data.lua [hunk 3]
```diff
@@ -378,6 +378,7 @@ local jail = function(player, griefer, msg)
 
     game.players[griefer].clear_console()
     Utils.print_to(griefer, message)
+    game.players[griefer].opened = none
```
Comment (TheBigZet) [inline utils/datastore/jail_data.lua:381]: Sorry, please change it to 
`game.players[griefer].opened = defines.gui_type.none`
so it's syntax correct

Comment (TheBigZet) [inline utils/datastore/jail_data.lua:381]: Sorry for nitpicking, please change it to 
`game.players[griefer].opened = defines.gui_type.none`
so it's syntax correct

**Review by TheBigZet — CHANGES_REQUESTED** [review]

File: utils/datastore/jail_data.lua (hunk 3 above)
Comment (Ragnarok77-factorio) [reply utils/datastore/jail_data.lua]: No worries it's fine, and it's done  @TheBigZet

**Review by TheBigZet — APPROVED** [review]

**Pull request merged**